tests/bearlib/languages/documentation/DocumentationCommentTest.py

```diff
@@ -36,3 +39,97 @@ def test_fields(self):
+
+    def load_testdata(self, filename):
+        filename = (os.path.dirname(os.path.realpath(__file__)) +
+                    "/documentation_extraction_testdata/" + filename)
+        with open(filename, "r") as fl:
+            data = fl.read()
+
+        return data.splitlines(keepends=True)
+
+    def test_empty_docstring(self):
+        self.check_docstring("", [])
+
+    def test_description(self):
+        doc = " description only "
+        self.check_docstring(doc, [self.Description(desc=' description only ')])
+
+    def test_params_default(self):
+        self.maxDiff = None
+        doc = (" :param test:  test description1 \n"
+               " :param test:  test description2 \n")
+        expected = [self.Parameter(name='test', desc=' test description1 \n'),
+                    self.Parameter(name='test', desc=' test description2 \n')]
+        self.check_docstring(doc, expected)
+
+    def test_return_values_default(self):
+        doc = (" :return: something1 \n"
+               " :return: something2 ")
+        expected = [self.ReturnValue(desc='something1 \n'),
+                    self.ReturnValue(desc='something2 ')]
+        self.check_docstring(doc, expected)
+
+    def test_python_default(self):
+        data = self.load_testdata("default.py")
+
+        language = "python"
+        docstyle = "default"
+
+        docs = tuple(extract_documentation(data, language, docstyle))
+
+        parsed_docs = []
+
+        for doc in docs:
+            parsed_docs.append(doc.parse())
+
+        expected = [
+            [self.Description(desc='\nModule description.\n\n'
+                                   'Some more foobar-like text.\n')],
+            [self.Description(desc='\nA nice and neat way of '
+                                   'documenting code.\n'),
+             self.Parameter(name='radius', desc='The explosion radius.\n')],
+            [self.Description(desc='\nA function that returns 55.\n')],
+            [self.Description(desc='\nDocstring with layouted text.\n\n    '
+                                   'layouts inside docs are preserved.'
+                                   '\nthis is intended.\n')],
+            [self.Description(desc=' Docstring inline with triple quotes.\n'
+                                   '    Continues here. ')],
+            [self.Description(desc='\nThis is the best docstring ever!\n'),
+             self.Parameter(name='param1:',
+                            desc='    Very Very Long Parameter description.\n'),
+             self.Parameter(name='param2:',
+                            desc='    Short Param description.\n'),
+             self.ReturnValue(desc='Long Return Description That Makes No Sense'
+                              ' And Will\n         Cut to the Next Line.\n')]]
+
+        self.assertEqual(parsed_docs, expected)
+
+    def check_docstring(self,
+                        docstring,
+                        expected=[]):
```

Comment:
would you move this function to the top? as this is an assertion function and provides functionality to the unittest class it's easier to read tests if they follow such custom check functions :+1: